### ISSUE TITLE
StaggeredMotion API shouldn't onlyChild

### ISSUE THREAD

**Issue by appsforartists** [chenglou/react-motion#257]
It feels natural to return `styles.map(…)` from StaggeredMotion's child function.  However, if you do that you'll get bit by the frustratingly unhelpful `onlyChild` assertion.  `onlyChild` can be hard to pin down in a standard React app, but there are so many layers of abstraction/nesting in the StaggeredMotion API that where exactly onlyChild is being violated is even harder to catch.

Why can't StaggeredMotion's child return an Array?  Can we skip the `onlyChild` here?

If not, StaggeredMotion should have a `tagName` attribute that accepts an HTML tag name or a ReactComponent that it will wrap your Array in.  That would eliminate a level of nesting and make the resulting StaggeredMotion call easier to read.

If StaggeredMotion can't return an Array and you don't like that `tagName` API proposal, can we please use a more explicit error message?  For instance, "StaggeredMotion's render function must return exactly one child, but it received an Array.  Please wrap the array in a container."

**Comment by chenglou** [comment]
See #20 and #44

Tldr: there's no extra level of nesting (unless you're talking about the visual one), since we return directly the result of your children function as the return value of `render`. We also avoid the extra prop `tagName` since you return your own tag.

I wish we use enough of Flow that this would be a non-issue. Alas, we're not there yet. So yeah we should add a warning.

**Comment by chenglou** [comment]
Turned out the invariant message is plenty. It does warn against accidentally returning an array: 
<img width="606" alt="screenshot 2016-04-19 15 55 44" src="https://cloud.githubusercontent.com/assets/1909539/14658246/31d7e234-0647-11e6-8f3c-ee51726bc08d.png">

Closing this for now. If people keep tripping on it I'll reopen and add a custom check!
